zfs-prune-snapshots

```diff
@@ -8,6 +8,9 @@
 
 VERSION='v1.1.0'
 
+# get PATH
+PATH="/usr/local/sbin:/usr/local/bin:/usr/sbin:/usr/bin:/sbin:/bin"
```

Comment:
I don't think a users `PATH` should be overridden by a script or program.  If `zfs` isn't in your path, then that is an environment issue most likely.  Some distros have this stored in `/etc/environment`, or if you login over ssh as `~/.ssh/environment`.

Having this set by the script is not the route I'd want to take.

Reply:
I see. I used to use `source /etc/environment` but was told off by a colleague because "it is not portable." :)
The environmental issue exist only when running from cron (or non-login terminals in general) since there are no environment variables inherited. I see no harm in using a local PATH variable since it exists only in the spawned shell instance and does not mess with the user's PATH outside.
I'll reconsider the environent file!

Reply:
> I don't think a users `PATH` should be overridden by a script or program. If `zfs` isn't in your path, then that is an environment issue most likely. Some distros have this stored in `/etc/environment`, or if you login over ssh as `~/.ssh/environment`.

Alpine Linux does not ship `/etc/environment` by default as its pam specific, so if zfs-prune-snapshots will source it, we need to check first that it exists.

Ref: sudo bug related to   `/etc/environment`  https://bugzilla.sudo.ws/show_bug.cgi?id=907

Reply:
Oh, let me clarify: I'm not advocating that `zfs-prune-snapshots` source any OS or distro-specific environment file.  I'm of the mindset that scripts shouldn't specify their own path, and instead should inherit the path from their environment (whoever calls the script, be it a person or a non-interactive login).

For example, `cron` sets `PATH` (and other environmental variables) to whatever is inherited from its own environment.  What I mean is, when cron executes a cronjob, the environment is copied from the environment of the cron daemon itself.  On my own personal machines, I make sure that cron is started with the environment that I want.

Some versions of cron, for example, allow specifying env vars over the command line of the config like:

``` cron
0 1 * * * PATH=/foo:/bar:/baz my_command arg1 arg2 ...
```

Alternatively, I would say that a wrapper script could be created for `zfs-prune-snapshots` like:

``` bash
#!/usr/bin/env bash
export PATH=whatever-you-want
exec zfs-prune-snapshots "$@"
```

And then you could call this from cron and customize it however you please. 

---

I know I've said a lot here.  My main point is that I believe that `PATH` (and other env vars) are the responsibility of the caller - whether that be a human on a shell, or a task kicked off via cron, and `zfs-prune-snapshots` should operate by simply calling `zfs`, and not trying to impose its own view of what the environment should look like.